src/spork.cpp

```diff
@@ -88,41 +127,60 @@ void CSporkManager::ProcessSpork(CNode* pfrom, const std::string& strCommand, CD
             if(!chainActive.Tip()) return;
             strLogMsg = strprintf("SPORK -- hash: %s id: %d value: %10d bestHeight: %d peer=%d", hash.ToString(), spork.nSporkID, spork.nValue, chainActive.Height(), pfrom->id);
         }
+
+        CKeyID keyIDSigner;
+        if (!(spork.GetSignerKeyID(keyIDSigner, IsSporkActive(SPORK_6_NEW_SIGS))
+                                 && setSporkPubKeyIDs.count(keyIDSigner))) {
+            // Note: unlike for other messages we have to check for new format even with SPORK_6_NEW_SIGS
```

Comment:
It was an error, `GetSignerKeyID` can't process this issue really, because `CPubKey::RecoverCompact` will return true and restore some key with any flag value. Without this additional check daemon starts to ban all its peers,  that's why I've added it. I'll remove not working code from `GetSignerKeyID`.